megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -2098,6 +2105,55 @@ private void unloadTransportedInfantry(MovePath path) {
         }
     }
     
+    /**
+     * Helper function that adds an "launch" step for units that are transporting 
+     * launchable units in some kind of bay.
+     */
+    private void launchFighters(MovePath path) {
+        // if my objective is to cross the board, even though it's tempting, I won't be leaving the infantry
+        // behind. They're not that good at screening against high speed pursuit anyway.
+        if(getBehaviorSettings().shouldGoHome()) {
+            return;
+        }
+        
+        Entity movingEntity = path.getEntity();
+        Coords pathEndpoint = path.getFinalCoords();
+        Targetable closestEnemy = getPathRanker(movingEntity).findClosestEnemy(movingEntity, pathEndpoint, getGame(), false);
+
+        // if there are no enemies on the board, then we're not launching anything.
+        if((null == closestEnemy) || (closestEnemy.getTargetType() != Targetable.TYPE_ENTITY)) {
+            return;
+        }
+        
+        TreeMap<Integer, Vector<Integer>> unitsToLaunch = new TreeMap<>();
```

Comment:
Can just be a map

Reply:
Nope, it's passed to a method that expects a TreeMap